plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryPlugin.java

```diff
@@ -165,6 +171,33 @@ static Settings getAvailabilityZoneNodeAttributes(Settings settings, String azMe
         return attrs.build();
     }
 
+    @SuppressForbidden(reason = "We call getInputStream in doPrivileged and provide SocketPermission")
+    private static Optional<String> getToken(String azMetadataTokenUrl) {
+        if (Strings.isNullOrEmpty(azMetadataTokenUrl)) {
+            return Optional.empty();
+        }
+        HttpURLConnection tokenUrlConnection;
+        try {
+            tokenUrlConnection = SocketAccess.doPrivilegedIOException(
+                () -> (HttpURLConnection) new URL(azMetadataTokenUrl).openConnection()
+            );
+            tokenUrlConnection.setConnectTimeout(CONNECT_TIMEOUT);
+            tokenUrlConnection.setRequestProperty("X-aws-ec2-metadata-token-ttl-seconds", String.valueOf(METADATA_TOKEN_TTL_SECONDS));
+        } catch (IOException e) {
+            logger.warn("Unable to access the IMDSv2 URI: " + azMetadataTokenUrl, e);
+            return Optional.empty();
+        }
+        try (
+            var in = SocketAccess.doPrivilegedIOException(tokenUrlConnection::getInputStream);
```

Comment:
I think we need to close this InputStream once we're done with it to?

Reply:
I think it should be automatically closed since it's in the try declaration block.

Reply:
ah sorry, I need more ☕